src/storage/wal/catalog_delta_entry.cppm

```diff
@@ -86,15 +90,16 @@ public:
     explicit AddDBMetaOperation(TxnTimeStamp begin_ts, bool is_delete, String db_name, String data_dir)
         : CatalogDeltaOperation(begin_ts, is_delete), db_name_(std::move(db_name)), data_dir_(std::move(data_dir)) {}
     explicit AddDBMetaOperation(DBMeta *db_meta) : CatalogDeltaOperation(CatalogDeltaOperationType::ADD_DATABASE_META), db_meta_(db_meta) {}
-    CatalogDeltaOperationType GetType() final { return CatalogDeltaOperationType::ADD_DATABASE_META; }
+    CatalogDeltaOperationType GetType() const final { return CatalogDeltaOperationType::ADD_DATABASE_META; }
     [[nodiscard]] SizeT GetSizeInBytes() const final {
         auto total_size =
             sizeof(CatalogDeltaOperationType) + sizeof(i32) + this->db_name_.size() + sizeof(i32) + this->data_dir_.size() + GetBaseSizeInBytes();
         return total_size;
     }
     void WriteAdv(char *&buf) const final;
     void SaveSate() final;
-    const String ToString() const final { return String("AddDBMetaOperation"); }
+    const String ToString() const final { return String("ADD_DATABASE_META"); }
+    const String EncodeIndex() const final { return String(fmt::format("{}_{}", i32(GetType()), this->db_name_)); }
```

Comment:
'_' is valid in various name. So '_' might not be a good delimiter of different item of the string. Try to selection another delimiter, such as '#', '/' or '|'.